### PR TITLE
Bugfix: The entire clan is no longer poisoned by gen_hunt_strangerabbit1

### LINKED ISSUE #2432: [BUG] All cats being poisoned when gen_hunt_strangerabbit1 fails
**Grade**: (only select one)
Important

**Describe the bug**
When when gen_hunt_strangerabbit1 fails, all cats in the Clan will be poisoned.

**Reproduce**
1. Go on a hunting patrol
2. Get patrol gen_hunt_strangerabbit1
3. Fail the patrol and have cats be poisoned
4. All cats in the Clan will be poisoned [ See error ]

**Patrol ID + Outcome Print (if applicable)**:
PATROL START ---------------------------------------------------
Patrol Leader: Wrenprance
Random Cat: Thornpaw
chosen filter prey size: large
Total Number of Possible Patrols | normal: 5, romantic: 0 
---
Finding stat cat. Outcome Type: Success = True, Antag = False
Can Have Stat: []
POSSIBLE STAT CATS ['Wrenprance']
No Stat Cat Found
---
starting chance: 30 | EX_updated chance: 36
Wrenprance updated chance to 36 | Thornpaw updated chance to 36 | 
PATROL ID: gen_hunt_strangerabbit1 | SUCCESS: False
PATROL END -----------------------------------------------------

**Commit # or Game Version Number**: 
Development Version 7c874d4d47debc08084be7dc7e0b210b9d495908

**Reproduce**
1. Go on a hunting patrol
2. Get patrol gen_hunt_strangerabbit1

**Additional context**
A code developer will comment the additional context.

**Screenshots**
The entire Clan being poisoned. I unfortunately don't have a screenshot of the patrol result, though.
Screenshot 2024-06-11 at 8 01 18 AM

### PR DESCRIPTION
**Pull request by j-gynn**
## About The Pull Request

A possibly-stopgap fix to stop the entire Clan being poisoned by the strangerabbit1 fail event by removing the `some_clan` tag.

Currently, `some_clan` and `clan` have the exact same backend code. I don't know what the intended outcome for `some_clan` is, if it's not this behaviour. It appears to be an undocumented abbreviation? What is its intended behaviour?

## Linked Issues
Fixes #2432

## Proof of Testing

image
image
(the story specifically requests that r_c is affected, meaning the minimum number of affected cats is 2)

### CODE REVIEW

File: scripts/utility.py [hunk 1]
```diff
@@ -1275,7 +1275,8 @@ def gather_cat_objects(
         elif abbr == "clan":
             out_set.update([x for x in Cat.all_cats_list if not (x.dead or x.outside or x.exiled)])
         elif abbr == "some_clan":
-            out_set.update([x for x in Cat.all_cats_list if not (x.dead or x.outside or x.exiled)])
+            out_set.update([x for x in Cat.all_cats_list if
+                            not (x.dead or x.outside or x.exiled)])  # are these supposed to be the same??
```
Comment (j-gynn) [inline scripts/utility.py]:
```suggestion
            clan_cats = [x for x in Cat.all_cats_list if not (x.dead or x.outside or x.exiled)]
            out_set.update(sample(clan_cats, randint(1, round(len(clan_cats) / 4))))
```
This is a possible implementation that returns up to a quarter of the clan's cats. I don't know what we want to happen here, fwiw, but I wanted to note a possible solution

**Comment by scribblecrumb** [comment]
Intended behavior is close to what you suggested, honestly.  It's essentially that "some" of the cats in the clan get affected, and that those cats are just chosen randomly.  I'm surprised I didn't notice that it wasn't working when I did the refactor haha.  

Frankly, it wouldn't be bad for us to split the "some_clan" tag into more specific options (ala the supply reduce tags, maybe?  so "clan_eighth", "clan_fourth", "clan_half".  and then affect that given fraction of the clan).  As right now it's not *super* useful for a wide range of things of situations.  However, it's OG intent was literally just to allow this specific patrol to poison a random selection of clanmates, so for now we can just tailor it's effect to that intent.  I think bringing your suggested implementation down to selecting 1/8th of the cats in the clan, rather than 1/4, would be closer to the intended effect for that patrol.

**Comment by j-gynn** [comment]
Like this?

**Comment by scribblecrumb** [comment]
Looks good!  Add a proof of testing and I'll get this approved.

**Comment by j-gynn** [comment]
> Looks good! Add a proof of testing and I'll get this approved.

Done :)

**Review by scribblecrumb — APPROVED** [review]

**Pull request merged**